### PR TITLE
fix(picker): picker-quiet-layout

### PR DESCRIPTION
**Pull request by blunteshwar**
## Description

Added a new property called `sideLabel` to `sp-picker` to introduce a new UI for sideLabel picker variant.

## Related issue(s)

- fixes https://github.com/adobe/spectrum-web-components/issues/2963

## How has this been tested?

-   [ ] _Test case 1_
    1. Go [here](https://blunteshwar-picker-bug--spectrum-web-components.netlify.app/storybook/?path=/story/picker--side-label)
-   [ ] _Test case 2_
    1. Go [here](https://blunteshwar-picker-bug--spectrum-web-components.netlify.app/storybook/?path=/story/picker--quiet-side-label)

## Screenshots (if appropriate)
Screenshot 2024-02-21 at 5 22 50 PM
Screenshot 2024-02-21 at 5 22 37 PM

## Types of changes

-   [] Bug fix (non-breaking change which fixes an issue)
-   [x] New feature (non-breaking change which adds functionality)
-   [ ] Breaking change (fix or feature that would cause existing functionality to change)
-   [ ] Chore (minor updates related to the tooling or maintenance of the repository, does not impact compiled assets)

## Checklist

-   [x] I have signed the [Adobe Open Source CLA](http://opensource.adobe.com/cla.html).
-   [x] My code follows the code style of this project.
-   [x] If my change required a change to the documentation, I have updated the documentation in this pull request.
-   [x] I have read the **[CONTRIBUTING](<(https://github.com/adobe/spectrum-web-components/blob/main/CONTRIBUTING.md)>)** document.
-   [ ] I have added tests to cover my changes.
-   [x] All new and existing tests passed.
-   [x] I have reviewed at the Accessibility Practices for this feature, see: [Aria Practices](https://www.w3.org/TR/wai-aria-practices/)

## Best practices

This repository uses conventional commit syntax for each commit message; note that the GitHub UI does not use this by default so be cautious when accepting suggested changes. Avoid the "Update branch" button on the pull request and opt instead for rebasing your branch against `main`.

### CODE REVIEW

**Comment by Rajdeepc** [comment]
@blunteshwar If you can add some tests for this change that would be great!

File: packages/picker/src/Picker.ts [hunk 1]
```diff
@@ -433,7 +433,7 @@ export class PickerBase extends SizedMixin(Focusable, { noDefaultSize: true }) {
                 aria-haspopup="true"
                 aria-labelledby="icon label applied-label"
                 id="button"
-                class="button"
+                class=${this.quiet ? 'button sideLabel' : 'button'}
```
Comment (Rajdeepc) [inline packages/picker/src/Picker.ts:436]: You can simplify by writing like this:
```
class=${`button ${this.quiet ? 'sideLabel' : ''}`}
```

**Review by Rajdeepc — COMMENTED** [review]
Few comments and also please add in few tests

File: packages/picker/src/Picker.ts (hunk 1 above)
Comment (Westbrook) [inline packages/picker/src/Picker.ts:436]: What about something being `quiet` also makes it have a "side label"?

Comment (blunteshwar) [reply packages/picker/src/Picker.ts:436]: I think this is only specific to `picker`

Comment (blunteshwar) [reply packages/picker/src/Picker.ts:436]: I talked to @jenndiaz in spectrum-css and she told me that the might rename this `side-label` class to a more generic class name. And yes, we will have to to use this class

**Comment by Westbrook** [comment]
It would be great if there wasn't a requirement for both the label and the picker to notate the label location. In that way, when this is "done" then the story decorator should "just work", rather than look like: 
image

I shared an possible path forward on this with @TarunAdobe on Tuesday, can yall sync up on that?

**Comment by Rajdeepc** [comment]
> It would be great if there wasn't a requirement for both the label and the picker to notate the label location. In that way, when this is "done" then the story decorator should "just work", rather than look like: image
> 
> I shared an possible path forward on this with @TarunAdobe on Tuesday, can yall sync up on that?

I think the above code is a little misleading. The side label of picker is adjusted with `display:flex` on the parent which is not something we should be doing all the time. This way the picker will not know that it is rendering a side label.
Instead, we should allow the `side-aligned` attribute on `sp-field-label` associated with a picker and target the picker to let it self know that it should allow the `sideLabel` attribute on it to remove any depedency of users adding this attribute from their side. Your thoughts?

**Comment by Westbrook** [comment]
> ```
>if (this.hasAttribute('side-aligned')) {
>             this.resolvedElement.element?.setAttribute('sideLabel', 'true');
>  }
> ```

Roughly, yes. This versions applied this value to everything, which may be right, but may not happen with the same attribute/property over time. What do you think about altering this type to include a second argument that is the entire Field Label class?

```diff
- applyFocusElementLabel?: (label?: string) => void;
+ applyFocusElementLabel?: (appliedLabel: string, labelElement?: FieldLabel) => void;
```

Then we can leverage all of the data on the FieldLabel as we see fit in the labeled element. Something like:

```js
applyFocusElementLabel = (value: string, labelElement: FieldLabel): void => {
    this.appliedLabel = value;
    if (labelElement.sideAligned === 'start') {
       ...
    }
};
```

**Comment by Rajdeepc** [comment]
> ```diff
> FieldLabel
> ```

Yes! I would agree with this approach which would definitely detach the logic from the Field Label class. Thanks

**Pull request merged**